internal/exec/shell_utils.go

```diff
@@ -334,3 +352,34 @@ func execTerraformShellCommand(
 	u.LogDebug(atmosConfig, fmt.Sprintf("Exited shell: %s\n", state.String()))
 	return nil
 }
+
+// mergeEnvVars combines two sets of environment variables, with overrideEnv taking precedence.
+//
+// This is necessary because:
+//  1. We need to preserve existing system environment variables (PATH, HOME, etc.)
+//  2. Atmos-specific variables (TF_CLI_ARGS, ATMOS_* vars) must take precedence
+//  3. For conflicts, such as TF_CLI_ARGS_*, we need special handling to ensure proper merging rather than simple overwriting
```

Comment:
Per (3), we should move this here:

https://github.com/cloudposse/atmos/pull/898/files#r1906125932